lib/rust/lexer/definition/src/token.rs

```diff
@@ -510,11 +853,63 @@ mod tests {
     }
 
     #[test]
-    fn construct_text_token() {
-        let token = Token::Text("some prose goes here",0);
-        assert_shape(&token,Shape::text("some prose goes here"));
-        assert_length(&token,20);
-        // TODO [AA] Make this internally account for length of quotes.
+    fn construct_text_line_token() {
+        let token = Token::TextLine(TextStyle::RawLine,vec![],0);
+        assert_shape(&token,Shape::text_line(TextStyle::RawLine,vec![]));
+        assert_length(&token,2);
+    }
+
+    #[test]
+    fn construct_text_inline_block_token() {
+        let token = Token::TextInlineBlock(TextStyle::RawInlineBlock,vec![],0);
+        let shape = Shape::text_inline_block(TextStyle::RawInlineBlock,vec![]);
+        assert_shape(&token,shape);
+        assert_length(&token,3);
+    }
+
+    #[test]
+    fn construct_text_block_token() {
```

Comment:
TBH I dont know if all these `construct_...` tests are needed if we have just lexer tests instead. These construction mechanisms are rather very striaghtforward and it just adds a lot of "boilerplate" to be kept up to date